sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -547,4 +552,23 @@ class SessionCatalog(
     functionRegistry.lookupFunctionBuilder(name)
   }
 
+
+  // -----------------
+  // | Other methods |
+  // -----------------
+
+  /**
+   * Drop all existing databases (except "default") along with all associated tables,
+   * partitions and functions, and set the current database to "default".
+   */
+  def reset(): Unit = {
+    val default = "default"
+    listDatabases().filter(_ != default).foreach { db =>
+      dropDatabase(db, ignoreIfNotExists = false, cascade = true)
+    }
+    tempTables.clear()
+    functionRegistry.clear()
```

Comment:
Since a `FunctionRegistry` also hold all builtin functions, is it call for removing all builders or just removing all non-builtin builders?